packages/snaps-rpc-methods/src/permitted/scheduleBackgroundEvent.ts

```diff
@@ -0,0 +1,149 @@
+import type { MethodHooksObject } from '../utils';
+
+const methodName = 'snap_scheduleBackgroundEvent';
+
+const hookNames: MethodHooksObject<ScheduleBackgroundEventMethodHooks> = {
+  scheduleBackgroundEvent: true,
+  hasPermission: true,
+};
+
+type ScheduleBackgroundEventHookParams = {
+  date: string;
+  request: CronjobRpcRequest;
+};
+
+export type ScheduleBackgroundEventMethodHooks = {
+  scheduleBackgroundEvent: (
+    snapEvent: ScheduleBackgroundEventHookParams,
+  ) => string;
+
+  hasPermission: (permissionName: string) => boolean;
+};
+
+export const scheduleBackgroundEventHandler: PermittedHandlerExport<
+  ScheduleBackgroundEventMethodHooks,
+  ScheduleBackgroundEventParameters,
+  ScheduleBackgroundEventResult
+> = {
+  methodNames: [methodName],
+  implementation: getScheduleBackgroundEventImplementation,
+  hookNames,
+};
+
+const offsetRegex = /Z|([+-]\d{2}:?\d{2})$/u;
+const ScheduleBackgroundEventsParametersStruct = object({
+  date: refine(string(), 'date', (val) => {
+    const date = DateTime.fromISO(val);
+    if (date.isValid) {
+      // luxon doesn't have a reliable way to check if timezone info was not provided
+      if (!offsetRegex.test(val)) {
+        return 'ISO8601 string must have timezone information';
+      }
+      return true;
+    }
+    return 'Not a valid ISO8601 string';
+  }),
+  request: CronjobRpcRequestStruct,
+});
+
+export type ScheduleBackgroundEventParameters = InferMatching<
+  typeof ScheduleBackgroundEventsParametersStruct,
+  ScheduleBackgroundEventParams
+>;
+
+/**
+ * The `snap_scheduleBackgroundEvent` method implementation.
+ *
+ * @param req - The JSON-RPC request object.
+ * @param res - The JSON-RPC response object.
+ * @param _next - The `json-rpc-engine` "next" callback. Not used by this
+ * function.
+ * @param end - The `json-rpc-engine` "end" callback.
+ * @param hooks - The RPC method hooks.
+ * @param hooks.scheduleBackgroundEvent - The function to schedule a background event.
+ * @param hooks.hasPermission - The function to check if a snap has the `endowment:cronjob` permission.
+ * @returns An id representing the background event.
+ */
+async function getScheduleBackgroundEventImplementation(
+  req: JsonRpcRequest<ScheduleBackgroundEventParameters>,
+  res: PendingJsonRpcResponse<ScheduleBackgroundEventResult>,
+  _next: unknown,
+  end: JsonRpcEngineEndCallback,
+  {
+    scheduleBackgroundEvent,
+    hasPermission,
+  }: ScheduleBackgroundEventMethodHooks,
+): Promise<void> {
+  const { params } = req;
+
+  if (!hasPermission(SnapEndowments.Cronjob)) {
+    return end(providerErrors.unauthorized());
+  }
+
+  try {
+    const validatedParams = getValidatedParams(params);
+
+    const { date, request } = validatedParams;
+
+    // Make sure any millisecond precision is removed.
+    const truncatedDate = DateTime.fromISO(date, { setZone: true })
+      .startOf('second')
+      .toISO({
+        suppressMilliseconds: true,
+      }) as string;
```

Comment:
Why this type cast?

Reply:
Return type is `string | null`, I believe the `toISO` might return null for a non-ISO8601 string, but we're fine here since we do validation above.

Reply:
Alternatively we could just assert that the value exists, that should make TS happy